gtsfm/utils/rotation.py

```diff
@@ -0,0 +1,38 @@
+"""Utility functions for rotations.
+
+Authors: Ayush Baid
+"""
+from typing import Dict, List, Tuple
+
+import networkx as nx
+import numpy as np
+from gtsam import Rot3
+
+
+def random_rotation() -> Rot3:
+    """Sample a random rotation by generating a sample from the 4d unit sphere."""
+    q = np.random.randn(4) * 0.03
+    # make unit-length quaternion
+    q /= np.linalg.norm(q)
+    qw, qx, qy, qz = q
+    return Rot3(qw, qx, qy, qz)
+
+
+def initialize_global_rotations_using_mst(num_images: int, i2Ri1_dict: Dict[Tuple[int, int], Rot3]) -> List[Rot3]:
+    # Create a graph from the relative rotations dictionary
+    graph = nx.Graph()
+    for i1, i2 in i2Ri1_dict.keys():
+        # TODO: use inlier count as weight
+        graph.add_edge(i1, i2, weight=1)
```

Comment:
without a weight, it's not a MST, but just a ST. Is this intended?

Reply:
Yes right now I do not pipe in any additional information into Shonan. Eventually we can use some form of weight here.